src/main/scala/org/scalajs/tools/tsimporter/Importer.scala

```diff
@@ -195,14 +195,21 @@ class Importer(val output: java.io.PrintWriter) {
     }
   }
 
-  private def processPropertyDecl(owner: ContainerSymbol, name: Name,
+  private def processPropertyDecl(enclosing: ContainerSymbol, owner: ContainerSymbol, name: Name,
       tpe: TypeTree, modifiers: Modifiers, protectName: Boolean = true) {
     if (name.name != "prototype") {
       tpe match {
         case ObjectType(members) if members.forall(_.isInstanceOf[CallMember]) =>
           // alternative notation for overload methods - #3
           for (CallMember(signature) <- members)
             processDefDecl(owner, name, signature, protectName)
+        case ObjectType(members) =>
+          val module = enclosing.getModuleOrCreate(owner.name)
+          module.isGlobal = false
+          val classSym = module.getClassOrCreate(name)
```

Comment:
I think that if you swap the first character of `name` for its `toUppercase` version (basically `name.head.toUppercase + name.tail`), you will get nicer types. Since the convention in JS/TS is to use `camelCase` for fields, most of the time, this should typically translate to a nice `CamelCase`, which will be idiomatic Scala for a type name.

Reply:
fixed at https://github.com/sjrd/scala-js-ts-importer/pull/57/commits/105391786e15eddeb89cc9e619816a87ff750380